src/Bridge.ts

```diff
@@ -838,6 +866,24 @@ export class Bridge {
                     this.connectionManager.push(connection);
                 }
             }
+
+            // If it's a power level event for a new room, we might want to create the setup widget.
+            if (this.config.widgets?.roomSetupWidget?.addOnInvite && event.type === "m.room.power_levels" && event.state_key === "" && !this.connectionManager.isRoomConnected(roomId)) {
+                log.debug(`${roomId} got a new powerlevel change and isn't connected to any connections, testing to see if we should create a setup widget`)
+                const plEvent = new PowerLevelsEvent(event);
+                const currentPl = plEvent.content.users?.[this.as.botUserId] || plEvent.defaultUserLevel;
+                const previousPl = plEvent.previousContent?.users?.[this.as.botUserId] || plEvent.previousContent?.users_default;
+                const requiredPl = plEvent.content.events?.["im.vector.modular.widgets"] || plEvent.defaultStateEventLevel;
```

Comment:
This is an explicit check to see if the PL for our bot user has changed from it's previous value, rather than whether it's just valid. The reason being is we don't want to spawn a new invite widget every time any user in the room gets a PL bump, we only want to invoke it when our user has a PL change.

Reply:
Right, there seems to be two checks here: one if it's changed (and fair play for it having its own code path), but the other half is the currentPl >= requiredPl bit – that's where I was wondering if reuse is in order, so that we don't duplicate the "im.vector.modular.widgets" and all its logic. I'm not insistent on it, and perhaps it's easier to trace this way anyway.

Reply:
Ah, that's an optimization. The call to check `userHasPowerLevelFor` requires pulling the state from the homeserver which is an extra request I didn't want to make simply for code cleanliness. In theory, we could use it, but it would suck a bit.